src/provers/stark.rs

```diff
@@ -0,0 +1,31 @@
+//! Stark prover trait.
+
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+
+use super::error::ProverError;
+use super::execution_artifacts::{AirPublicInput, CairoExecutionArtifacts};
+use crate::settings::SettingsProvider;
+
+#[async_trait]
+pub trait StarkProver: Send + Sync + Sized {
+    fn with_settings(settings_provider: &impl SettingsProvider) -> Result<Self, ProverError>;
+    async fn prove(&self, execution_artifacts: &CairoExecutionArtifacts) -> Result<StarkProverOutput, ProverError>;
```

Comment:
considering proving a block takes multiple minutes, should this function be refactored to return a prover_job_id and we add another function to check the status of the prover_job_id?